src/index.js

```diff
@@ -1,50 +1,81 @@
 // @flow
-import Reconciler from 'react-reconciler';
+
+import Reconciler, {
+  type HostConfig,
+  type OpaqueHandle
+} from 'react-reconciler';
 import getOwnerDocument from 'dom-helpers/ownerDocument';
 
 import Root from './Root';
 import * as DOMComponent from './DOMComponent';
-import type { HostConfig as HC } from './reconciler-types'; // Could be imported from react-reconciler once it exports
+import {
+  cacheHandleByInstance,
+  getInternalHandleFromInstance
+} from './DOMComponentTree';
 
 function createElement(type, props, rootContainerElement): Element {
   const ownerDocument = getOwnerDocument(rootContainerElement);
   let domElement = ownerDocument.createElement(type);
   return domElement;
 }
 
-const HostConfig: HC = {
+const hostConfig: HostConfig<
+  string, // T: component type
+  Props, // P: props
+  Element, // I: component instance
+  Text, // TI: component text instance
+  Element, // HI: Hydration Instance
+  Element | Text, // PI: Public instance
+  Element, // C: Container instance
+  any, // Child container instance
+  HostContext, // CX: Host context
+  Array<[string, any]> // PL: prepare update result
```

Comment:
FWIW Flow should be able to infer this. Although personally I do prefer an explicit annotation myself.